pkgs/lib.nix

```diff
@@ -16,7 +16,8 @@ let self = {
       ProtectKernelTunables = "true";
       ProtectKernelModules = "true";
       ProtectControlGroups = "true";
-      RestrictAddressFamilies = "AF_UNIX AF_INET AF_INET6";
+      # AF_NETLINK is required by network libraries used in bitcoind and lightning-pool
+      RestrictAddressFamilies = "AF_UNIX AF_INET AF_INET6 AF_NETLINK";
```

Comment:
You noticed that `RestrictAddressFamilies` has no effect on stable because when porting to NixOS unstable you lightning-pool didn't work and bitcoind continued to work because we had set it to `allowAnyProtocol`, correct? Then, why not add `AF_NETLINK` to lightning-pool only?

> Allowing AF_NETLINK for unprivileged services poses no security risk.

A quick google search tells me otherwise.

Concept ACK to the rest of the PR.

Reply:
`Bitcoind+zmq` without `AF_NETLINK` breaks right at startup.
`lightning-pool` starts successfully but breaks when `bitcoind` finishes IBD. This is visible only in the `regtest` scenario.
I speculated that other services might break in other situations which are not covered by our tests, so I whitelisted `AF_NETLINK` for all services. 
A common use of `AF_NETLINK` by unprivileged processes is to read routing information via the kernel's `rnetlink` socket. 

> A quick google search tells me otherwise.

AFAIK, there have been kernel bugs in the past, but there's no structural risk given the kernel functions correctly.
I'd be okay with service-specific whitelisting, but it might lead to delayed runtime failures with obscure error messages. What do you prefer?

Reply:
> What do you prefer?

I prefer service-specific whitelisting with the possibility of delayed runtime failures with obscure error messages

Reply:
Fixed in `services: add finer-grained address family restrictions`.
I've also added `AF_NETLINK` to the zmq clightning plugin, which failed with no error message at all.

Reply:
Thanks @erikarvstedt. A bit annoying indeed that all this functionality is untestable in master until NixOS stable gets whatever is necessary from unstable. I noticed you don't give `AF_NETLINK` to `lnd` which had it before. Is that intentional?

Reply:
Ah it's mentioned in the commit message.

Reply:
> Ah it's mentioned in the commit message.
Yes, a few seconds before your post I forced pushed the following addition to the commit msg:
```
lnd: lnd only runs a zmq client, not a server, therefore it requires
no additional address families.
```

This PR is currently being tested on unstable via https://github.com/fort-nix/nix-bitcoin/pull/344.